src/highlight.rs

```diff
@@ -61,17 +61,25 @@ fn prepare_line(s: &str, tab_len: u8, mask: Option<char>) -> Cow<'_, str> {
         None => {
             let tab = spaces(tab_len);
             let mut buf = String::new();
+            let mut col: usize = 0;
             for (i, c) in s.char_indices() {
                 if buf.is_empty() {
                     if c == '\t' {
                         buf.reserve(s.len());
                         buf.push_str(&s[..i]);
-                        buf.push_str(tab);
+                        let len = tab_len as usize - (col % tab_len as usize);
```

Comment:
Panics for `tab_len` == 0

Reply:
@joshka Thank you for the point.

@pm100 Can you add a guard to avoid the panic?

```rust
if tab_len > 0 {
    let len = tab_len as usize - (col % tab_len as usize);
    buf.push_str(&tab[..len]);
    col += len;
}
```

Reply:
tabs cannot be entered if tab_len is 0 they are filtered out in insert_tab so this panic never occurs in real life.. I actually think thats incorrect for the hard case. Should I change it?

Reply:
Got it. That's the problem with testing internals rather than from a more public interface :)
That said, a quick look suggests that this method is called from several places which don't have restrictions on the `tab_len`. Take a look at `LineHighlighter::into_spans()`, which blindly takes the value from the field which has no check.

A test that is more robust (by testing the effect rather than the implementation) might test the public interface `into_spans()` - I'm not sure whether that's necessary however.

Also note @rhysd's comment on https://github.com/rhysd/tui-textarea/issues/34 that you can run the tests using:
```shell
cargo test --features=search -- --skip src/lib.rs
```

Reply:
@pm100 Yes. Tab characters can be included in the initial text passed at `TextArea::from`. We still need to consider it.